app/views/Project/Tagging/Sources/SourcesFilter/types.ts

```diff
@@ -0,0 +1,118 @@
+import {
+    AnalysisFrameworkFilterType as AnalysisFrameworkFilterTypeRaw,
+} from '#generated/types';
+
+type BaseFilterType = Omit<AnalysisFrameworkFilterTypeRaw, 'properties'>;
+
+export interface KeyLabelEntity {
+    // FIXME: clientId should be renamed to be key
+    clientId: string;
+    label: string;
+    tooltip?: string;
+    order: number;
+}
+
+export interface KeyLabel {
+    key: string;
+    label: string;
+}
+
+interface BaseProperties<T> {
+    type: string;
+    options?: T[];
+}
```

Comment:
Just found out yesterday that properties can be undefined, so we should handle cases where properties are undefined and also define their types as nullable